framework/base/src/abi/type_abi_impl_codec_multi.rs

```diff
@@ -75,16 +75,16 @@ impl<T: TypeAbi> TypeAbi for OptionalValue<T> {
 }
 
 macro_rules! multi_arg_impls {
-    ($(($mval_struct:ident $($n:tt $name:ident)+) )+) => {
+    ($(($mval_struct:ident $($n:tt $t:ident $u:ident)+) )+) => {
         $(
-            impl<$($name),+ > TypeAbiFrom<Self> for crate::codec::multi_types::$mval_struct<$($name,)+>
+            impl<$($t, $u),+ > TypeAbiFrom<crate::codec::multi_types::$mval_struct<$($u,)+>> for crate::codec::multi_types::$mval_struct<$($t,)+>
             where
-                $($name: TypeAbi,)+
+                $($t: TypeAbiFrom<$u>,)+
             {}
 
-            impl<$($name),+ > TypeAbi for crate::codec::multi_types::$mval_struct<$($name,)+>
+            impl<$($t),+ > TypeAbi for crate::codec::multi_types::$mval_struct<$($t,)+>
             where
-                $($name: TypeAbi,)+
+                $($t: TypeAbi,)+
             {
                 type Unmanaged = Self;
```

Comment:
also implement `type  Unmanaged = $mval_struct<$($name::Unmanaged,)+*> `
and add a test